### PR TITLE
Upgrade to Fastify v3

### PR DESCRIPTION
**Pull request by thomheymann**
Upgrade to Fastify v3

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 'use strict'
 
 const t = require('tap')
-const test = t.test
```
Comment (mcollina) [inline test.js:4]: This would be handy to retain.

File: test.js [hunk 2]
```diff
@@ -308,41 +317,34 @@ test('Expose status route with additional route options', t => {
...
+
+  const fastify = Fastify()
+
   fastify.addHook('onRoute', (routeOptions) => {
     fastify.server.unref()
     process.nextTick(() => block(500))
-    t.strictEqual(routeOptions.url, '/alive')
+    t.strictEqual(routeOptions.url, '/status')
     t.strictEqual(routeOptions.logLevel, 'silent', 'log level not set')
-    t.deepEqual(routeOptions.config, customConfig, 'config not set')
```
Comment (mcollina) [inline test.js:316]: why was this removed?

File: test.js [hunk 3]
```diff
@@ -308,41 +317,34 @@ test('Expose status route with additional route options', t => {
...
-  fastify.addHook('onRoute', (routeOptions) => {
-    fastify.server.unref()
-    process.nextTick(() => block(500))
-    t.strictEqual(routeOptions.url, '/status')
-    t.strictEqual(routeOptions.logLevel, 'silent', 'log level not set')
-    fastify.close()
-  })
 
-  fastify.listen()
+  fastify.ready()
```
Comment (mcollina) [inline test.js:344]:
```suggestion
  await fastify.ready()
```

File: index.d.ts [hunk 4]
```diff
@@ -11,21 +17,23 @@ declare namespace underPressure {
...
     sampleInterval?: number;
-    exposeStatusRoute?: boolean | string | {routeOpts: object; url?: string};
+    exposeStatusRoute?: boolean | string | { routeOpts: object; url?: string };
   }
 }
 
 declare module "fastify" {
-  interface FastifyInstance<HttpServer, HttpRequest, HttpResponse> {
-    memoryUsage: () => string;
+  interface FastifyInstance<
```
Comment (Ethan-Arrowood) [inline index.d.ts:25]: I don't think you need to define these generics here.

File: package.json [hunk 5]
```diff
@@ -26,19 +26,20 @@
   "author": "Tomas Della Vedova - @delvedor (http://delved.org)",
   "license": "MIT",
   "dependencies": {
-    "fastify-plugin": "^1.6.1"
+    "fastify-plugin": "^2.0.0"
   },
   "devDependencies": {
     "@types/node": "^13.13.2",
-    "fastify": "^2.13.1",
+    "fastify": "^3.0.0-alpha.1",
```
Comment (Ethan-Arrowood) [inline package.json:33]: use `rc`

**Comment by mcollina** [comment]
Done in https://github.com/fastify/under-pressure/pull/51
